### PR TITLE
Updated touch plugin with separated gesture handler

### PR DESCRIPTION
**Pull request by merceavanessa**

### CODE REVIEW

File: jquery.flot.touch.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        function shutdown(plot, eventHolder) {
+            eventHolder[0].removeEventListener('touchstart', interpretGestures);
+            eventHolder[0].removeEventListener('touchmove', interpretGestures);
+            eventHolder[0].removeEventListener('touchend', interpretGestures);
+        }
+
+        var pan = {
+            touchstart: function(e) {
+                preventEventPropagation(e);
+                plot.getPlaceholder()[0].childNodes[2].dispatchEvent(new CustomEvent('panstart', { detail: e }));
```
Comment (cristiingineru) [inline jquery.flot.touch.js]: The order of the nodes can change. When bindEvents is called store the eventHolder argument in a member variable and reuse it here.

File: jquery.flot.touch.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                plot.getPlaceholder()[0].childNodes[2].dispatchEvent(new CustomEvent('pinchdrag', { detail: e }));
+            },
+
+            touchend: function(e) {
+                preventEventPropagation(e);
+            }
+        };
+
+        var doubleTap = {
+            recenterPlot: function(e) {
```
Comment (cipix2000) [inline jquery.flot.touch.js]: rename this to something better

**Comment by cipix2000** [comment]
please add the new file to .npmignore so it is included in distribution

File: jquery.flot.touchNavigate.js [hunk 3]
```diff
@@ -230,4 +130,156 @@
...
+    }
+
+    function distance(x1, y1, x2, y2) {
+        return Math.sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
+    }
+
+    function pinchDistance(e) {
+        var t1 = e.detail.touches[0],
+            t2 = e.detail.touches[1];
+        return distance(t1.pageX, t1.pageY, t2.pageX, t2.pageY);
```
Comment (cristiingineru) [inline jquery.flot.touchNavigate.js]: The pan and pinch events are 'high level' events which should make this plug-in's life easier by avoiding dealing with low level details. If possible the pan or pinch events should contain extra things like... the pinch distance.

plot.getPlaceholder()[0].childNodes[2].dispatchEvent(new CustomEvent('pinchstart', { detail: e, pinchDistance: 123 }));

**Pull request merged**